connector/connect/server/src/main/scala/org/apache/spark/sql/connect/service/ExecuteEventsManager.scala

```diff
@@ -278,6 +278,7 @@ case class ExecuteEventsManager(executeHolder: ExecuteHolder, clock: Clock) {
  * @param extraTags:
  *   Additional metadata during the request.
  */
+@JsonIgnoreProperties({ "planRequest" })
```

Comment:
@gengliangwang @rednaxelafx does something need to be done so that history service defaults None on this planRequest missing from json, or will this just work?
Are there any suites where event log replay from json is tested where a test could be added for this?

Reply:
I think this just works IIRC.

Reply:
For testing, simply doing ser/de test should be enough, see also https://github.com/apache/spark/pull/23728 in which I fixed a similar problem.

Reply:
I added json serde testing as done in https://github.com/apache/spark/pull/23728 and`@JsonIgnoreProperties({ "planRequest" })` didn't work
```
[info] - SPARK-43923: post started *** FAILED *** (2 seconds, 220 milliseconds)
[info]   com.fasterxml.jackson.databind.exc.InvalidDefinitionException: Cannot find a deserializer for non-concrete Collection type [collection type; class com.google.protobuf.ProtocolStringList, contains [simple type, class java.lang.String]]
[info]  at [Source: (String)"{"Event":"org.apache.spark.sql.connect.service.SparkListenerConnectOperationStarted","jobTag":"SparkConnect_OperationTag_User_7f8102fa-64a6-4c6e-a2a1-d701de9c0dc3_Session_1_Operation_6a771009-bc23-4767-bef7-6f75cf92b3a0","operationId":"6a771009-bc23-4767-bef7-6f75cf92b3a0","eventTime":0,"sessionId":"7f8102fa-64a6-4c6e-a2a1-d701de9c0dc3","userId":"1","userName":"userName","statementText":"limit {\n  limit: 10\n}\n","sparkSessionTags":[],"extraTags":{}}"; line: 1, column: 1]
[info]   at com.fasterxml.jackson.databind.exc.InvalidDefinitionException.from(InvalidDefinitionException.java:67)
[info]   at com.fasterxml.jackson.databind.DeserializationContext.reportBadDefinition(DeserializationContext.java:1915)
[info]   at com.fasterxml.jackson.databind.deser.DeserializerCache._createAndCache2(DeserializerCache.java:268)
[info]   at com.fasterxml.jackson.databind.deser.DeserializerCache._createAndCacheValueDeserializer(DeserializerCache.java:244)
[info]   at com.fasterxml.jackson.databind.deser.DeserializerCache.findValueDeserializer(DeserializerCache.java:142)
[info]   at com.fasterxml.jackson.databind.DeserializationContext.findNonContextualValueDeserializer(DeserializationContext.java:644)
[info]   at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.resolve(BeanDeserializerBase.java:539)
[info]   at com.fasterxml.jackson.databind.deser.DeserializerCache._createAndCache2(DeserializerCache.java:294)
[info]   at com.fasterxml.jackson.databind.deser.DeserializerCache._createAndCacheValueDeserializer(DeserializerCache.java:244)
[info]   at com.fasterxml.jackson.databind.deser.DeserializerCache.findValueDeserializer(DeserializerCache.java:142)
[info]   at com.fasterxml.jackson.databind.DeserializationContext.findContextualValueDeserializer(DeserializationContext.java:621)
[info]   at com.fasterxml.jackson.module.scala.deser.OptionDeserializer.createContextual(OptionDeserializerModule.scala:50)
[info]   at com.fasterxml.jackson.databind.DeserializationContext.handleSecondaryContextualization(DeserializationContext.java:867)
[info]   at com.fasterxml.jackson.databind.DeserializationContext.findContextualValueDeserializer(DeserializationContext.java:623)
[info]   at com.fasterxml.jackson.databind.deser.impl.PropertyBasedCreator.construct(PropertyBasedCreator.java:114)
[info]   at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.resolve(BeanDeserializerBase.java:636)
[info]   at com.fasterxml.jackson.databind.deser.DeserializerCache._createAndCache2(DeserializerCache.java:294)
[info]   at com.fasterxml.jackson.databind.deser.DeserializerCache._createAndCacheValueDeserializer(DeserializerCache.java:244)
[info]   at com.fasterxml.jackson.databind.deser.DeserializerCache.findValueDeserializer(DeserializerCache.java:142)
[info]   at com.fasterxml.jackson.databind.DeserializationContext.findRootValueDeserializer(DeserializationContext.java:654)
[info]   at com.fasterxml.jackson.databind.ObjectMapper._findRootDeserializer(ObjectMapper.java:4956)
[info]   at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4826)
[info]   at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3772)
[info]   at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3740)
[info]   at org.apache.spark.util.JsonProtocol$.sparkEventFromJson(JsonProtocol.scala:877)
```
. 
I moved to `@JsonIgnore`. Unfortunately ,the constructor is no longer compatible, but the blast radius should be limited as most users of this API should be readers.